### ISSUE TITLE
[BUG] ImportError: cannot import name 'TabularDataset' from 'autogluon.tabular' (unknown location)

### ISSUE THREAD

**Issue by cheldieva-l** [autogluon/autogluon#4422]
Kaggle notebook cpu:
 I run the installation and get an error when importing:

```python
pip install torch==2.3.1 torchvision==0.18.1 --index-url https://download.pytorch.org/whl/cpu
!git clone https://github.com/autogluon/autogluon
!cd autogluon && ./full_install.sh
from autogluon.tabular import TabularDataset, TabularPredictor
```

https://www.kaggle.com/code/arabidopsisthalian/dataset-int-autogluon-as-is-3cmd
```python
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[8], line 1
----> 1 from autogluon.tabular import TabularDataset, TabularPredictor

ImportError: cannot import name 'TabularDataset' from 'autogluon.tabular' (unknown location)
```

https://www.kaggle.com/code/arabidopsisthalian/dataset-int-autogluon-as-is-5cmd
```python
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[38], line 1
----> 1 from autogluon.tabular import TabularDataset, TabularPredictor

ImportError: cannot import name 'TabularDataset' from 'autogluon.tabular' (unknown location)
```

**Bug Report Checklist**
  I provided code that demonstrates a minimal reproducible example. 
Kaggle notebook cpu:
pip install torch==2.3.1 torchvision==0.18.1 --index-url https://download.pytorch.org/whl/cpu
!git clone https://github.com/autogluon/autogluon
!cd autogluon && ./full_install.sh
from autogluon.tabular import TabularDataset, TabularPredictor

https://www.kaggle.com/code/arabidopsisthalian/dataset-int-autogluon-as-is-5cmd
https://www.kaggle.com/code/arabidopsisthalian/dataset-int-autogluon-as-is-3cmd

- [X ] I confirmed bug exists on the latest mainline of AutoGluon via source install. 
- [ ] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**

**Expected behavior**

**To Reproduce**

**Screenshots / Logs**

**Installed Versions**

<details>

```python
# Replace this code with the output of the following:
from autogluon.core.utils import show_versions
show_versions()
```

</details>

**Comment by Innixma** [comment]
Thanks for the report! Seems likely that this might have to do with recent changes to some of our `pyproject.toml` files due to migrating to `ruff` (#4412).

For example, in the Kaggle notebook I can import classes via `from autogluon.common import FeatureMetadata`, but not `from autogluon.tabular import TabularPredictor`.

I'd need to test a bit further to get into the details.

In the meantime, you can use the nightly pre-release via `pip install --pre autogluon`, which should be very close to a source install in terms of functionality.

**Comment by Innixma** [comment]
@cheldieva-l The following works:

```
!pip install --pre "autogluon==1.1.2b20240815"
```

**Comment by Innixma** [comment]
Here is a reproducible example on Colab of the bug: https://colab.research.google.com/drive/11MV_gAlkiWVOBh5M7b_C4NkM9j0Koc8J?usp=sharing

**Comment by Innixma** [comment]
@cheldieva-l This is fixed in #4425, you can do the following:

```
!git clone -b fix_source_install https://github.com/autogluon/autogluon
!cd autogluon && ./full_install.sh

from autogluon.tabular import TabularDataset, TabularPredictor
```

**Comment by Innixma** [comment]
@cheldieva-l PR #4425 has been merged. You should be able to run your notebook as is (with restart runtime after install) and it should now work.
